time_test.go

```diff
@@ -0,0 +1,65 @@
+package pflag
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+)
+
+func setUpTime(t *time.Time, formats []string) *FlagSet {
+	f := NewFlagSet("test", ContinueOnError)
+	f.TimeVar(t, "time", time.Time{}, formats, "Time")
+	return f
+}
+
+func TestTime(t *testing.T) {
+	testCases := []struct {
+		input    string
+		success  bool
+		expected string
+	}{
+		{"2022-01-01T01:01:01+00:00", true, "2022-01-01T01:01:01Z"},
+		{" 2022-01-01T01:01:01+00:00", true, "2022-01-01T01:01:01Z"},
+		{"2022-01-01T01:01:01+00:00 ", true, "2022-01-01T01:01:01Z"},
+		{"2022-01-01T01:01:01+02:00", true, "2022-01-01T01:01:01+02:00"},
+		{"2022-01-01T01:01:01.01+02:00", true, "2022-01-01T01:01:01.01+02:00"},
+		{"Sat, 01 Jan 2022 01:01:01 +0000", true, "2022-01-01T01:01:01Z"},
+		{"Sat, 01 Jan 2022 01:01:01 +0200", true, "2022-01-01T01:01:01+02:00"},
+		{"Sat, 01 Jan 2022 01:01:01 +0000", true, "2022-01-01T01:01:01Z"},
+		{"", false, ""},
+		{"not a date", false, ""},
+		{"2022-01-01 01:01:01", false, ""},
+		{"2022-01-01T01:01:01", false, ""},
+		{"01 Jan 2022 01:01:01 +0000", false, ""},
+		{"Sat, 01 Jan 2022 01:01:01", false, ""},
+	}
+
+	devnull, _ := os.Open(os.DevNull)
+	os.Stderr = devnull
```

Comment:
Why is this needed? What in this test would otherwise write to stderr?

Reply:
For this one going back by memory I believe only reason I've added it was because of copy paste from the test file I modified to make it easier to stick to the repo style.

That is one of these
- https://github.com/search?q=repo%3Aspf13%2Fpflag+%09os.Stderr+%3D+devnull&type=code

I've checked now and as far as I can tell non of the occurrences of this in the repo are doing anything.
Removed it for time_test.go in d15848db482b52179577da9738cf9702d8d28466

Reply:
Thanks for the quick response to the review request 🙇